backend/src/main/java/com/bakdata/conquery/apiv1/QueryProcessor.java

```diff
@@ -377,4 +388,46 @@ public ExternalUploadResult uploadEntities(Subject subject, Dataset dataset, Ext
 				statistic.getUnreadableDate()
 		);
 	}
+
+	/**
+	 * Execute a {@link TableExportQuery} for a single Entity on some Connectors.
+	 *
+	 * @implNote we don't do anything special here, this request could also be made manually. We however want to encapsulate this behaviour to shield the frontend from knowing too much about the query engine.
+	 */
+	public Response getSingleEntityExport(Subject subject, String idKind, String entity, List<Connector> sources, String format, Dataset dataset, Range<LocalDate> dateRange) {
+
+		final ConceptQuery
+				entitySelectQuery =
+				new ConceptQuery(new CQDateRestriction(dateRange, new CQExternal(List.of(idKind), new String[][]{{"HEAD"}, {entity}})));
+
+		final TableExportQuery exportQuery = new TableExportQuery(entitySelectQuery);
+		exportQuery.setTables(
+				sources.stream()
+					   .map(source -> new CQUnfilteredTable(source, null))
+					   .collect(Collectors.toList())
+		);
+
+		final ManagedQuery execution = (ManagedQuery) postQuery(dataset, exportQuery, subject);
+
+		// collect id immediately so it does not get sucked into closure
+		final ManagedExecutionId id = execution.getId();
+
+		while (execution.awaitDone(10, TimeUnit.SECONDS) == ExecutionState.RUNNING) {
+			log.trace("Still waiting for {}", id);
+		}
+
+		// Use the provided format name to find the respective provider.
+		final ResultRendererProvider rendererProvider =
+				config.getResultProviders().stream()
+					  .filter(provider -> provider.getClass().getAnnotation(CPSType.class).id().equalsIgnoreCase(format))
+					  .findFirst()
+					  .orElseThrow(() -> new BadRequestException(String.format("No configured provider for `%s` found.", format)));
+
+		final Response response = rendererProvider.createResult(
+				subject, execution, dataset, true, determineCharset(null, null),
+				() -> storage.removeExecution(id)
+		);
+
+		return response;
```

Comment:
Wie besprochen: Rückgabe sollten die ResultUrls sein nicht das Ergebnis direkt

Reply:
Von der Absprache habe ich leider nichts mitbekommen, beim nächsten mal bitte in einem entsprechenden Ticket dokumentieren, wenn Absprachen ohne mich über mein Tickets getroffen werden.

Die Anforderung ist für mich im Grunde in Ordnung, ich frage mich nur, wie wir die Anfragen sinnvoll von den Usern verstecken und/oder ggf nach dem herunterladen löschen. Das werden nämlich schnell echt viele werden, deswegen hab ich die Hook zum löschen nach dem herunterladen da eingebaut.

Würde ein Flag in den Download Endpunkten was ein löschen nach dem herunterladen veranlasst reichen?

Reply:
> Würde die Anfrage denn dem User momentan angezeigt werden? Wird die nicht noch rausgefiltert?

Die gehört ja dem User also sollten die die auch sehen können (halt nicht expandieren). Wäre ein `system` flag, was auch persistenz blockiert hier vlt schon die Lösung?

Reply:
Ich habs mal relativ rough umgesetzt als system Flag, was ausgabe ind er API blockiert, werden dann beim QueryCleanup gelöscht